src/Automation/Automation/Common/AutomationPSClientDSC.cs

```diff
@@ -981,7 +981,24 @@ public CompilationJob StartCompilationJob(string resourceGroupName, string autom
 
                 };
 
-                var job = this.automationManagementClient.DscCompilationJob.Create(resourceGroupName, automationAccountName, Guid.NewGuid().ToString(), createJobParameters);
+                string jobId = Guid.NewGuid().ToString();
+
+                var jobTask = this.automationManagementClient.DscCompilationJob.CreateAsync(resourceGroupName, automationAccountName, jobId, createJobParameters).GetAwaiter();
```

Comment:
The correct way to implement this is using [BeginCreate](https://github.com/Azure/azure-sdk-for-net/blob/master/src/SDKs/Automation/Management.Automation/Generated/DscCompilationJobOperationsExtensions.cs#L248)  or [BeginCreateAsync](https://github.com/Azure/azure-sdk-for-net/blob/master/src/SDKs/Automation/Management.Automation/Generated/DscCompilationJobOperationsExtensions.cs#L275) - the Begin* calls return as soon as the 'Accept' is issued to the initial request, rather than polling for status

Reply:
Added the wait to return the information of the job as, in some of my tests, it was returning too soon. Plus, CreateAsync doesn't turn info of the job itself. That's the reason why this code is implemented like that. Make sense?

Reply:
Regarding the test, I would assume there should be already a test case for this, I haven't checked honestly because this is a fix on a specific condition.

Reply:
Both Create and CreateAsync issue the request to start the job and then begin polling.  In the case of CreateAsync, the polling occurs inside the Task, so you are using up a thread doing polling that is unnecessary.

BeginCreate and BeginCreateAsync simply make the REST call to initiate the Start command, and do not poll on status.

Reply:
Gotcha. Thanks for the clarification. Removed the extra wait loop and switched to use BeginCreate method instead